addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRule.java

```diff
@@ -450,40 +413,173 @@ public int getWascId() {
         return 15; // WASC-15: Application Misconfiguration
     }
 
-    private void raiseAlert(
-            String name,
-            String otherInfo,
-            String param,
-            int risk,
-            String evidence,
-            String alertRef) {
-        raiseAlert(name, otherInfo, param, risk, evidence, getReference(), alertRef);
+    private AlertBuilder getBuilder() {
+        return newAlert()
+                .setConfidence(Alert.CONFIDENCE_HIGH)
+                .setDescription(Constant.messages.getString(MESSAGE_PREFIX + "desc"))
+                .setSolution(getSolution())
+                .setReference(getReference())
+                .setCweId(getCweId())
+                .setWascId(getWascId());
     }
 
-    private void raiseAlert(
-            String name,
-            String otherInfo,
-            String param,
-            int risk,
-            String evidence,
-            String refs,
-            String alertRef) {
-        String alertName = StringUtils.isEmpty(name) ? getName() : getName() + ": " + name;
-
-        newAlert()
-                .setName(alertName)
+    private AlertBuilder buildXcspAlert(int risk, String evidence, String param) {
```

Comment:
Worth making the order consistent in all methods, in some places the `evidence` is before `param` and in others after.